mirascope/core/base/_structured_stream.py

```diff
@@ -73,7 +74,11 @@ def __iter__(self) -> Generator[_ResponseModelT, None, None]:
                 yield extract_tool_return(self.response_model, json_output, True)
         if json_output:
             json_output = json_output[: json_output.rfind("}") + 1]
-        yield extract_tool_return(self.response_model, json_output, False)
+        extracted_response_model = extract_tool_return(
+            self.response_model, json_output, False
+        )
+        self.constructed_response_model = extracted_response_model
+        yield extracted_response_model
```

Comment:
> can you explain why we need this? I'm assuming it's so that the user can easily grab the final constructed object after the stream without having to keep track of it in their own stream?
correct
> Regardless, we should just do self.constructed_response_model = extract_tool_return(...) and then yield self.constructed_response_model no need for the double assignment
done